nomad/csi_endpoint.go

```diff
@@ -1118,6 +1182,112 @@ func (v *CSIVolume) createVolume(vol *structs.CSIVolume, plugin *structs.CSIPlug
 	return nil
 }
 
+// expandVolume validates the requested capacity values and issues
+// ControllerExpandVolume (and NodeExpandVolume, if needed) to the CSI plugin,
+// via Nomad client RPC.
+//
+// Note that capacity can only be increased; reduction in size is not possible,
+// and if the volume is already at the desired capacity, no action is taken.
+// vol Capacity-related values are mutated if successful, so callers should
+// pass in a copy, then commit changes to raft.
+func (v *CSIVolume) expandVolume(vol *structs.CSIVolume, plugin *structs.CSIPlugin, capacity *csi.CapacityRange) error {
+	if vol == nil || plugin == nil || capacity == nil {
+		return errors.New("unexpected nil value")
+	}
```

Comment:
this should never happen -- this is my fear of panics showing.  I actually wonder if a panic would be preferable? because at least then it'd provide a full stack trace if there is an unexpected bug...